docs/services/versioning.mdx

```diff
@@ -4,22 +4,42 @@ description: "Understand deployments, registration, and versioning in Restate."
 icon: "send-backward"
 ---
 
-For durable execution to work correctly, Restate must know exactly which version of your code to use for each request. This requires understanding three key concepts: **deployments**, **registration**, and **versioning**.
+**Service versioning is critical for durable execution.** When requests can sleep for hours or days, Restate needs to ensure they always use the same code version they started with.
+When Restate resumes an invocation, it replays the journal. The journal should be executed against the same code version to ensure deterministic behaviour.
 
-## What is a Deployment?
+This requires understanding two fundamental approaches: **immutable versions** and **in-place updates**.
 
-A deployment in Restate is a specific, versioned instance of your service code—whether running as an HTTP endpoint, a Lambda function, or another supported environment.
+## Immutable versions vs In-place updates
 
-Each deployment is immutable: once registered, its code and endpoint must not change.
+**Immutable versions** create a completely new deployment for each code change. Once a deployment is registered with Restate, its code cannot change. Restate automatically proxies new requests to the latest version, while ongoing requests continue with their original version until completion.
 
-An invocation is bound to a specific deployment — it starts and completes within that same deployment. This ensures that in-flight requests always see the code they started with, preserving correctness and determinism.
+**In-place updates** modify the code at an existing endpoint without creating a new deployment. While sometimes necessary for critical bug fixes, this approach can break determinism if not handled carefully.
 
-## Registering a Deployment
+**We strongly recommend using immutable versions for simplicity.** They guarantee that your durable executions remain deterministic without requiring you to think about the complexities of breaking existing in-flight invocations.
```

Comment:
I would make this part more assertive on the following fact: we support only **Immutable versions**, but we give you escape hatches for mutating deployments in place if you need to.

Things like:

> While sometimes necessary for critical bug fixes, this approach can break determinism if not handled carefully.

IMO is not assertive enough. **this approach break determinism** without can.

> **We strongly recommend using immutable versions for simplicity.**

This should also be something like Restate supports immutable deployments out of the box, they guarantee blablabla